src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,11 +1,15 @@
 package com.epam.processor;
```

Comment:
General point for entire file - all methods with 7 should be implemented with Java 7 api, ie without streaming. in this way you will see what's the difference is and will be able to decide by yourself in future what to choose.